### PR TITLE
MCR-2009 mods authority type is ignored

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2009).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 8d4751f61a7da222d8e7ecf652015f92c1659f9f! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4224-pull_request-752

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 8d4751f61a7da222d8e7ecf652015f92c1659f9f! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4224-pull_request-752

**Review by yagee-de — CHANGES_REQUESTED** [review]
This should do the trick
```
diff --git a/mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassMapper.java b/mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassMapper.java
index 849202de6..21ee43d48 100644
--- a/mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassMapper.java
+++ b/mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassMapper.java
@@ -135,9 +135,7 @@ public class MCRClassMapper {
                 authInfo = new MCRAuthorityWithURI(authorityURI, valueURI);
             } else if (modsElement.getAttributeValue("authority") != null) {
                 //authority
-                String authority = modsElement.getAttributeValue("authority");
-                String code = modsElement.getTextTrim();
-                authInfo = new MCRAuthorityAndCode(authority, code);
+                authInfo = MCRAuthorityAndCode.getAuthorityInfo(modsElement);
             } else if (modsElement.getName().equals(ACCESS_CONDITION)) {
                 //accessDefinition
                 String href = modsElement.getAttributeValue("href", MCRConstants.XLINK_NAMESPACE, "");
@@ -164,9 +162,7 @@ public class MCRClassMapper {
                 authInfo = new MCRAuthorityWithURI(authorityURI, valueURI);
             } else if (!modsElement.getAttribute("authority").isEmpty()) {
                 //authority
-                String authority = modsElement.getAttribute("authority");
-                String code = modsElement.getTextContent().trim();
-                authInfo = new MCRAuthorityAndCode(authority, code);
+                authInfo = MCRAuthorityAndCode.getAuthorityInfo(modsElement);
             } else if (modsElement.getLocalName().equals(ACCESS_CONDITION)) {
                 //accessDefinition
                 String href = modsElement.getAttributeNS(MCRConstants.XLINK_NAMESPACE.getURI(), "href");
```

File: mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassMapper.java [hunk 1]
```diff
@@ -149,6 +147,9 @@ public static MCRCategoryID getCategoryID(org.jdom2.Element modsElement) {
             } else {
                 return null;
             }
+            if(authInfo==null){
```
Comment (yagee-de) [inline mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassMapper.java]: please check code style

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Review by yagee-de — APPROVED** [review]

**Pull request merged**